tests/src/utils/databaseUtil.test.ts

```diff
@@ -1,16 +1,108 @@
 import { expect } from 'chai';
-import { existsSync, readFileSync, writeFileSync, unlinkSync } from 'fs';
-import { defaultFilePath, databaseUtil } from '../../../src/utils/databaseUtil';
+import fs from 'fs';
+import sinon, { SinonSpy } from 'sinon';
+import {
+  defaultFilePath,
+  databaseUtil,
+  DeleteResult,
+  LoadResult,
+  SaveResult,
+} from '../../../src/utils/databaseUtil';
 import { testSingletonModule } from '../../testHelpers/semanticMocha';
 
 testSingletonModule('utils/databaseUtil', ({ testIntegration }) => {
+  testIntegration('delete', ({ testScenario }) => {
+    testScenario('when the file exists and it is deleted without error')
+      .arrange(() => {
+        fs.writeFileSync(defaultFilePath, '');
+      })
+      .act(() => {
+        const deleteResult = databaseUtil.delete();
+        const fileExists = fs.existsSync(defaultFilePath);
+
+        return {
+          deleteResult,
+          fileExists,
+        };
+      })
+      .assert('returns a success result', (arranged, { deleteResult }) => {
+        const expectedResult: DeleteResult = {
+          isFileOnDisk: false,
+          error: null,
+        };
+
+        expect(deleteResult).to.eql(expectedResult);
+      })
+      .assert('deletes the file', (arranged, { fileExists }) => {
+        expect(fileExists).to.eq(false);
+      });
+
+    testScenario('when the file does not exist')
+      .arrange(() => {
+        expect(fs.existsSync(defaultFilePath)).to.eq(false);
+      })
+      .act(() => databaseUtil.delete())
+      .assert('returns an error result', (arranged, deleteResult) => {
+        const expectedNormalizedResult: DeleteResult = {
+          isFileOnDisk: false,
+          error: "ENOENT: no such file or directory, unlink 'saves/data.json'",
+        };
+
+        const { error } = deleteResult;
+        expect(error).to.be.instanceof(Error);
+        expect({
+          ...deleteResult,
+          error: (error as Error).message,
+        }).to.eql(expectedNormalizedResult);
+      });
+
+    // Note: Ideally we don't want to stub fs, but I don't know how to simulate this scenario on Windows
```

Comment:
I think this is fine for now. One way we could test this case would be change the file permissions to cause the `fs.unlinkSync` to fail.

Reply:
I spent a suuuuuuuuper long time trying to do that on Windows, but the file permissions are a lot different than linux. I made [this ticket](https://github.com/normal-devs/klicker-knight/projects/1#card-77267526) to attempt to write an integration test that only runs on certain operating systems

Reply:
It's going to be fun!